web/src/main/java/org/springframework/security/web/UnreachableFilterChainException.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web;
+
+/**
+ * Thrown if {@link SecurityFilterChain securityFilterChain} is not valid.
+ *
+ * @author Max Batischev
+ * @since 6.5
+ */
+public class UnreachableFilterChainException extends IllegalArgumentException {
```

Comment:
Sorry I didn't mention this in the last review, it's a bit of a new process. Since exceptions are serializable, they are subject to our serialization process.

Please do the following:

1. Add `private static final int serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;` to the class
2. Activate and run `SpringSecurityCoreVersionSerializableTests#serializeCurrentVersionClasses`
3. Add the resulting `.serialized` file to the PR